dataactvalidator/sqs_work_dispatcher.py

```diff
@@ -0,0 +1,706 @@
+import logging
+import inspect
+import json
+import os
+
+import psutil as ps
+import signal
+import sys
+import time
+import multiprocessing as mp
+
+from botocore.exceptions import ClientError, EndpointConnectionError, NoCredentialsError, NoRegionError
+
+from dataactcore.aws.sqsHandler import sqs_queue
+from dataactvalidator.validator_logging import log_job_message
+
+# Not a complete list of signals
+# Only the following are allowed to be used with signal() on Windows
+#    SIGABRT, SIGFPE, SIGILL, SIGINT, SIGSEGV, SIGTERM, or SIGBREAK (windows only)
+# NOTE: 1-15 are relatively standard on unix platforms; > 15 can change from platform to platform
+BSD_SIGNALS = {
+        1: "SIGHUP [1] (Hangup detected on controlling terminal or death of controlling process)",
+        2: "SIGINT [2] (Interrupt from keyboard)",
+        3: "SIGQUIT [3] (Quit from keyboard)",
+        4: "SIGILL [4] (Illegal Instruction)",
+        6: "SIGABRT [6] (Abort signal from abort(3))",
+        8: "SIGFPE [8] (Floating point exception)",
+        9: "SIGKILL [9] (Non-catchable, non-ignorable kill)",
+        11: "SIGSEGV [11] (Invalid memory reference)",
+        14: "SIGALRM [14] (Timer signal from alarm(2))",
+        15: "SIGTERM [15] (Software termination signal)",
+        19: "SIGSTOP [19] (Suspend process execution)",  # NOTE: 17 on Mac OSX, 19 on RHEL
+        20: "SIGTSTP [20] (Interrupt from keyboard to suspend (CTRL-Z)",  # NOTE: 18 on Mac OSX, 20 on RHEL
+    }
+if sys.platform == "win32":
+    BSD_SIGNALS[0] = "CTRL_C_EVENT [0] (Quit detected on controlling terminal with CTRL-C)"
+    BSD_SIGNALS[1] = "CTRL_BREAK_EVENT [1] (Quit detected on controlling terminal with CTRL-BREAK)"
+    BSD_SIGNALS[21] = "SIGBREAK [21] (Quit detected on controlling terminal with CTRL-BREAK)"
+
+
+class SQSWorkDispatcher:
+    _logger = logging.getLogger(__name__)
```

Comment:
In many of the logging config examples I read through, it was considered safer to declare the logger within the class rather than at the module level, so that the logger is not instantiated until _after_ the logging configuration has been parsed and configured in the case of using `fileConfig()`, or `dictConfig()` to read the logging config. The reason is that each of `fileConfig()`, and `dictConfig() has a backwards compatible config param named `disabled_existing_loggers`, that is set to `True` by default. So if you import a module that declares its logger at the module level, and then configure logging _after_ the import (and hence after the instantiation of the module-level logger), e.g. in the `__main__` module as we do in `app.py`, that previously instantiated logger will get disabled. I believe this might be circumvented by also declaring a logger for that specific module in the config. And it could be circumvented by setting `disable_existing_loggers=False` in the config, but we don't currently do that. So, in general, it's more defensive to just delay instantiation of the logger until a point when you know the config is set -- either during class instantiation, or by getting a new logger instance each time you log.

See 
- *Warning* under: https://docs.python.org/3.5/howto/logging.html#configuring-logging, 
- and also https://fangpenlin.com/posts/2012/08/26/good-logging-practice-in-python/#do-not-get-logger-at-the-module-level-unless-disable_existing_loggers-is-false
- and comments about `disable_existing_loggers` in the *Update:* section of this SO post answered by the guy who wrote the Python logging module: https://stackoverflow.com/a/15735146